### PR TITLE
 add a sanitizer to properly remove the base url

### LINKED ISSUE #354: Add repo form is allows https, but stores it as undefined
When a user adds a repo using the full URL, it works but it saves the name as `undefined`.

The expectation is that undefined should be saved. Perha

Steps to fix:
- [ ] write a test that ensures that only the save data can only be the `owner/repo` format.
- [ ] add a sanitizer to properly remove the `https://github.com/`

Line that should be fixed is: https://github.com/bdougie/open-sauced/blob/master/src/components/AddRepoForm.js#L20

### PR DESCRIPTION
**Pull request by NileshPatel17**
1. added a method to check the validity of the url, and get relative repo url.
2. added test cases for the same.

Please suggest me if there is better way to do it. would appreciate if you can give me any suggestion/feedback on test cases.
Thanks!

closes #354

### CODE REVIEW

File: src/tests/AddRepoForm.test.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from "react";
 import "@testing-library/jest-dom/extend-expect";
-import {render, cleanup} from "@testing-library/react";
-import AddRepoForm from "../components/AddRepoForm";
+import {render, cleanup, fireEvent} from "@testing-library/react";
```
Comment (bdougie) [inline src/tests/AddRepoForm.test.js]: fireEvent is unused, can you remove

File: src/components/AddRepoForm.js [hunk 2]
```diff
@@ -5,22 +5,36 @@ import {CardPadding} from "../styles/Card";
...
+  const isFullUrl = regex.test(url);
+  const relativeRepoUrl = isFullUrl ? url.toLowerCase().replace(regex, "") : url.toLowerCase();
+
+  const [owner, repo] = relativeRepoUrl.split("/");
+  if (!owner || !repo) {
+    return [false, null];
+  }
+  return [true, relativeRepoUrl];
+}
+
```
Comment (bdougie) [inline src/components/AddRepoForm.js]: considering this doesn't have to be specific to this component. Can you move this function to a separate file into the lib folder?

File: src/tests/AddRepoForm.test.js [hunk 3]
```diff
@@ -21,3 +21,46 @@ test("should render a link to track first repository if there are no repositorie
...
+  test(`repo-url '${ownerName}' should fail`, () => {
+    const [isValid] = isValidRepoUrl(`${ownerName}`);
+    expect(isValid).toBe(false);
+  });
+  test("Case insensitive test", () => {
+    expect(isValidRepoUrl("Http://github.com/owner_name/repo_name")[0]).toBe(true);
+    expect(isValidRepoUrl("Https://github.com/owner_name/repo_name")[0]).toBe(true);
+    expect(isValidRepoUrl("http://GITHUB.com/owner_name/repo_name")[0]).toBe(true);
+  });
+});
```
Comment (bdougie) [inline src/tests/AddRepoForm.test.js]: Per my suggestion above, the test is not component-specific and can be moved to a separate file as well.

**Comment by bdougie** [comment]
Thanks so much for grabbing this issue. I left a few comments to reviews, once address it can be merged

File: src/tests/AddRepoForm.test.js [hunk 4]
```diff
@@ -21,3 +21,46 @@ test("should render a link to track first repository if there are no repositorie
   const {container} = render(<AddRepoForm />);
   expect(container).toHaveTextContent("add");
 });
+describe("Test: isValidRepoUrl()", () => {
+  const repoName = "repo_name";
+  const ownerName = "owner_name";
+  const baseUrl = "http://github.com/";
+  const baseSecueUrl = "https://github.com/";
```
Comment (bdougie) [inline src/tests/AddRepoForm.test.js]: typo: baseSecureUrl

File: src/components/AddRepoForm.js (hunk 2 above)
Comment (NileshPatel17) [reply src/components/AddRepoForm.js]: moved to lib/util.js. Hope this is fine.

File: src/tests/AddRepoForm.test.js (hunk 1 above)
Comment (NileshPatel17) [reply src/tests/AddRepoForm.test.js]: restore ```AddRepoForm.test.js```` as moved all test cases to ```tests/util.test.js```. Hence it is not required any more.

File: src/tests/AddRepoForm.test.js (hunk 3 above)
Comment (NileshPatel17) [reply src/tests/AddRepoForm.test.js]: Done. moved all test cases to ```tests/util.test.js```

File: src/tests/AddRepoForm.test.js (hunk 4 above)
Comment (NileshPatel17) [reply src/tests/AddRepoForm.test.js]: Fixed in new test file ```util.test.js```

**Pull request merged**